packages/contracts/deploy/30_upgrade_repo/_common.ts

```diff
@@ -62,56 +64,20 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
     `Upgrading from current protocol version v${current[0]}.${current[1]}.${current[2]} to the latest version v${latest[0]}.${latest[1]}.${latest[2]}.`
   );
 
-  // NOTE: The following code can be uncommented and `initData` can be filled
-  // with arguments in case re-initialization of the `PluginRepo` should become necessary.
-  // Re-initialization will happen through a call to `function initializeFrom(uint8[3] calldata _previousProtocolVersion, bytes calldata _initData)`
-  // that Aragon might add to the `PluginRepo` contract once it's required.
-  /*
-  // Define the `_initData` arguments 
-  const initData: BytesLike[] = [];
-  // Encode the call to `function initializeFrom(uint8[3] calldata _previousProtocolVersion, bytes calldata _initData)` with `initData`
-  const initializeFromCalldata =
-    latestPluginRepoImplementation.interface.encodeFunctionData('initializeFrom', [
-      current,
-      initData,
-    ]);
-  */
-  const initializeFromCalldata: BytesLike = [];
-
-  // Check if deployer has the permission to upgrade the plugin repo
-  if (
-    await pluginRepo.isGranted(
-      pluginRepo.address,
-      deployer.address,
-      PLUGIN_REPO_PERMISSIONS.UPGRADE_REPO_PERMISSION_ID,
-      []
-    )
-  ) {
-    // Use `upgradeToAndCall` if the new implementation must be re-initialized by calling
-    // on the `PluginRepo` proxy. If not, we use `upgradeTo`.
-    if (initializeFromCalldata.length > 0) {
-      await pluginRepo.upgradeToAndCall(
-        latestPluginRepoImplementation.address,
-        initializeFromCalldata
-      );
-    } else {
-      await pluginRepo.upgradeTo(latestPluginRepoImplementation.address);
-    }
-  } else {
-    throw Error(
-      `The new version cannot be published because the deployer ('${deployer.address}')
-      is lacking the ${PLUGIN_REPO_PERMISSIONS.UPGRADE_REPO_PERMISSION_ID} permission.`
-    );
-  }
-};
-export default func;
-func.tags = ['UpgradeRepo'];
+  return {
+    deployer,
+    pluginRepo,
+    latestPluginRepoImplementation,
+    current,
+    latest,
+  };
+}
 
 /**
- * Skips the plugin repo upgrade if exists in the plugin repo.
+ * Skips the plugin repo upgrade if the implementation is already up-to-date.
  * @param {HardhatRuntimeEnvironment} hre
  */
-func.skip = async (hre: HardhatRuntimeEnvironment) => {
+export const skipUpgrade = async (hre: HardhatRuntimeEnvironment) => {
```

Comment:
Separate issue than this PR but we could just as easily de-duplicate the code from the skip function and early return from the deploy function if the protocol version matches the current

Reply:
I don't understand the comment. Isn't this what's happening?

Reply:
Improved in 772c5fed49b3bc41e9cd9ac3c3d72ec542c0778d